Azure iot/device provisioning service

montgomp opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

Creating new autorest generated API for swagger spec:  [https://github.com/Azure/azure-rest-api-specs/pull/2093](https://github.com/Azure/azure-rest-api-specs/pull/2093)
<!--
Please add an informative description that covers that changes made by the pull request.

If you are regenerating your SDK based off of a new swagger spec, please add the link to the corresponding swagger spec pull request that has been merged in the azure-rest-api-specs repository
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

montgomp committed:
Recreate fork due to multiple issues in Azure/azure-sdk-for-net

shahabhijeet commented on src/SDKs/ProvisioningServices/ProvisioningServices.Tests/ProvisioningServices.Tests.csproj:
```diff
@@ -0,0 +1,45 @@
+  </ItemGroup>
+
+  <ItemGroup>
+    <!--<PackageReference Include="Microsoft.Azure.Management.ProvisioningServices" Version="1.1.0" />-->
```
@montgomp remove commented code

shahabhijeet commented on src/SDKs/ProvisioningServices/ProvisioningServices.Tests/ProvisioningServices.Tests.csproj:
```diff
@@ -0,0 +1,45 @@
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+  <ItemGroup>
```
@montgomp remove line 37 to 44.
all the references are being added to every test project

shahabhijeet commented on src/SDKs/ProvisioningServices/ProvisioningServices.Tests/ProvisioningServices.Tests.csproj:
```diff
@@ -0,0 +1,45 @@
+    <!--<PackageReference Include="Microsoft.Azure.Management.ProvisioningServices" Version="1.1.0" />-->
+
+    <ProjectReference Include="..\..\IotHub\Management.IotHub\Microsoft.Azure.Management.IotHub.csproj" />
+
```
@montgomp remove line break

shahabhijeet commented on src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj:
```diff
@@ -0,0 +1,24 @@
+    <Description>Provides management capabilities for Microsoft Azure IoT Device Provisioning Service.</Description>
+    <AssemblyTitle>Microsoft Azure Provisioning Service Management</AssemblyTitle>
+    <AssemblyName>Microsoft.Azure.Management.ProvisioningServices</AssemblyName>
+    <VersionPrefix>1.0.0</VersionPrefix>
```
@montgomp if this is your first package, please release your first preview package as 0.9.0-preview
Once you are ready to go stable, release it as 1.0.0 (without preview)
make similar changes in AssemblyInfo.cs

shahabhijeet reviewed (changes requested):
Can you do the following in the below order:
1)	Pull latest from upstream (if you have already done this good)
2)	Execute msbuild.exe build.proj from the repo root
a.	This will update the build tools in the current repo
3)	Generate your SDK
4)	Build you sdk at least one from command line of IDE
a.	Msbuild.exe build.proj /t:build /p:Scope=SDKs\ProvisioningServices
5)	Update the PR

montgomp committed:
Merge remote-tracking branch 'upstream/psSdkJson6' into IotHub/DeviceProvisioning

montgomp committed:
Update upstream per shahabhijeet.

Can you do the following in the below order:
Pull latest from upstream (if you have already done this good)
Execute msbuild.exe build.proj from the repo root
a. This will update the build tools in the current repo
Generate your SDK
Build you sdk at least one from command line of IDE
a. Msbuild.exe build.proj /t:build /p:Scope=SDKs\ProvisioningServices
Update the PR

montgomp committed:
PR review requested changes

montgomp committed:
Fix typo

montgomp commented on src/SDKs/ProvisioningServices/ProvisioningServices.Tests/Helpers/Constants.cs:
```diff
@@ -0,0 +1,42 @@
+
+        public static class Certificate
+        {
+            public const string Content =
```
@shahabhijeet I don't follow what you're recommending here. This is similar to implementation in use in tests for Management.IoTHub

Do you have a link to an example of proposed changes?

shahabhijeet commented on src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj:
```diff
@@ -0,0 +1,18 @@
+    <AssemblyName>Microsoft.Azure.Management.ProvisioningServices</AssemblyName>
+    <VersionPrefix>0.9.0-preview</VersionPrefix>
+    <PackageTags>Microsoft Azure IoT Provisioning Services;IoT Device Provisioning Service Management;IoT Provisioning Service;</PackageTags>
+  </PropertyGroup>
```
@montgomp please remove lines 10-17
Add packageReleaseNotes
Model your project looking at the other SDK project files. you are missing 1 imports at the bottom of your project file.

montgomp replied:
added the import statement. Moved PackageReleaseNotes node into PropertyGroup that contains other package information

montgomp replied:
Removing those lines will break compilation as it would remove a closing </PropertyGroup> and remove the TargetFrameworks node.

shahabhijeet commented on src/SDKs/IotHub/AzSdk.RP.props:
```diff
@@ -0,0 +1,7 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--This file and it's contents are updated at build time moving or editing might result in build failure. Take due deligence while editing this file-->
+  <PropertyGroup>
+    <AzureApiTag />
```
@montgomp why was this file created? Please undo this change.

montgomp replied:
done

montgomp replied:
@shahabhijeet 7 days ago you told me to remove this file. Today (2018/01/09) you tell me I need the AzSdk.RP.props file. I am unable to fulfill both requests.

shahabhijeet reviewed (changes requested):
Please generate using generate.cmd, that will create an additional meta data file needed for validating the PR.

montgomp committed:
Update to handle if unexpected access policy is used.

montgomp committed:
Feedback from shahabhijeet

montgomp committed:
Remove AzSdk.RP.props file randomly created

shahabhijeet commented:
@montgomp you need AzSdk.RP.props file, please include it

montgomp committed:
Regenerate client from latest swagger spec and rerecord tests.

montgomp commented:
@shahabhijeet the AzSdk.RP.props file that you asked me to recreate is the same one that you had asked me to remove earlier. It is being recreated automatically on each build and is in Microsoft.Azure.Management.IotHub assembly that is referenced by the test project. Please advise if you want the file included though no changes to IotHub are being performed as it is a separate assembly.

shahabhijeet commented on src/SDKs/ProvisioningServices/Management.ProvisioningServices/Microsoft.Azure.Management.ProvisioningServices.csproj:
```diff
@@ -0,0 +1,22 @@
+	<!-- Please do not move/edit code below this line -->
+	
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.ProvisioningServices</PackageId>
```
@montgomp any reason your service name in the swagger and your nuget package is not aligned.
For e.g. your nuget is called provisioning services, but your description uses Device Provisioning service and your API version in swagger is called just "devices"
This is just way too many names, your nuget name should be aligned with your service name.

montgomp replied:
I can change to Microsoft.Azure.Management.DeviceProvisioningService if that is the only thing holding up the PR. I cannot change the Microsoft.Devices in the swagger as that is not owned by my team. Please advise if this is acceptable as it is quite a large rework.

montgomp replied:
The swagger changes you suggest would be a breaking change for python CLI that is already released.

shahabhijeet replied:
The reason I have asked this is because, there is way for developers to search for a particular API version in nuget.org
So if your nuget is called ProvisioningService but your actual service is Devices, how will the developer know what to search for?
What is the official name of your service?

We cannot have different names for the service, it has to be consistent across.

montgomp replied:
The service name is "[Microsoft Azure IoT Hub Device Provisioning Service](https://docs.microsoft.com/en-us/azure/iot-dps/)". I think that is quite the marketing mouthful which is why I'm more inclined to namespace as Microsoft.Azure.Management.DeviceProvisioningService. 

Again, I cannot change the Microsoft.Devices references due to 1) not being a contributor to the various projects if you want to go down that rabbit hole. 2) The number of production breaks that would be caused to ARM, portal, CLI, and existing PowerShell cmdlets for IoT Hub and Device Provisioning Service. 

Are you suggesting I move this assembly away from the Microsoft.Azure.Management.XXX hierarchy that other service SDKs appear to use?

shahabhijeet commented:
@montgomp please reopen once you are ready to merge this.
This PR needs a fix in the spec, as soon as the spec PR is merged, you can either reopen or create a new PR.
